cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Query.hs

```diff
@@ -245,19 +258,22 @@ hprop_cli_queries = integrationWorkspace "cli-queries" $ \tempAbsBasePath' -> H.
     TestQueryRefScriptSizeCmd ->
       -- ref-script-size
       do
+        -- Set up files and vars
         refScriptSizeWork <- H.createDirectoryIfMissing $ work </> "ref-script-size-test"
         alwaysSucceedsSpendingPlutusPath <- File <$> liftIO (makeAbsolute "test/cardano-testnet-test/files/plutus/v3/always-succeeds.plutus")
         let transferAmount = 10_000_000
+        -- Submit a transaction to publish the reference script
         txBody <- buildTransferTx execConfig epochStateView sbe refScriptSizeWork "tx-body" wallet1
-                    [(ReferenceScriptAddress alwaysSucceedsSpendingPlutusPath, 10_000_000)]
+                    [(ReferenceScriptAddress alwaysSucceedsSpendingPlutusPath, transferAmount)]
         signedTx <- signTx execConfig cEra refScriptSizeWork "signed-tx" txBody [SomeKeyPair $ paymentKeyInfoPair wallet1]
         submitTx execConfig cEra signedTx
+        -- Wait until transaction is on chain and obtain transaction identifier
         txId <- retrieveTransactionId execConfig signedTx
-        -- wait for one block before checking for the transaction
-        _ <- waitForBlocks epochStateView 1
+        txIx <- H.evalMaybeM $ watchEpochStateUpdate epochStateView (EpochInterval 2) (getTxIx sbe txId transferAmount)
+        -- Query the reference script size
         let protocolParametersOutFile = refScriptSizeWork </> "ref-script-size-out.json"
         H.noteM_ $ execCli' execConfig [ eraName, "query", "ref-script-size"
-                                       , "--tx-in", txId ++ "#0"
+                                       , "--tx-in", txId ++ "#" ++ show (txIx :: Int)
```

Comment:
Were you getting intermittent failures due to specifying the wrong tx input?

Reply:
No, I think it is deterministic. The main reason I did it is minimizing the wait (finding out when it makes it on chain, instead of waiting for an arbitrary amount of blocks/epochs). But this is a nice side-effect, because it removes an assumption that is not at all obvious